src/librustc_resolve/lib.rs

```diff
@@ -3509,13 +3509,40 @@ impl<'a, 'tcx> Resolver<'a, 'tcx> {
                                         format!("to call `{}::{}`", path_str, path_name),
                                 };
 
+                                let mut help_msg = String::new();
                                 if !msg.is_empty() {
                                     msg = format!(". Did you mean {}?", msg)
+                                } else {
+                                    // we check if this a module and if so, we display a help
+                                    // message
+                                    let name_path = path.segments.iter()
+                                                        .map(|seg| seg.identifier.name)
+                                                        .collect::<Vec<_>>();
+                                    let current_module = self.current_module.clone();
+
+                                    match self.resolve_module_path(current_module,
+                                                   &name_path[..],
+                                                   UseLexicalScope,
+                                                   expr.span,
+                                                   PathSearch) {
+                                        Success(_) => {
+                                            help_msg = format!("A module called `{module}` is \
```

Comment:
Why not 

> To reference an item from the `{module}` module, use `{module}::{next_path_component}`.

instead? It is shorter and more-to-the-point. Also this obviously should become a `span_help`.

E.g. for your case it would then look like this:

> To reference an item from the `io` module, use `io::stdin`.